### ISSUE TITLE
Error: Cyclic Object Value

### ISSUE THREAD

**Issue by xrisdoc** [l-lin/angular-datatables#801]
I am obtaining data using a promise to render a DataTable and also want to include some buttons. However, when I include the buttons I seem to get the following error:

`Error: Cyclic object value`

The code I am using for setting the options are provided below:

```
this.dtInstance = {};

this.dtOptions = DTOptionsBuilder.fromFnPromise(function () {
    return $resource('/Api/MyData/', {}, {
        'getData': { method: 'POST', isArray: true }
    }).getData().$promise;
})
.withOption('processing', true)
.withPaginationType("full_numbers")
.withDisplayLength(25)
.withButtons(["colvis", "copy", "print", "excel"]); // Causes Error: Cyclic object value;
```

The table data does display within the table as expected and so do the buttons. However, due to this error, the **dtInstance** object is undefined, which I need to be able to access.

**Comment by xrisdoc** [comment]
I haven't got to the bottom of this issue yet, but I have managed to get it working in a slightly diffferent way without it causing that error.

The error seems to occur when the **dtInstance** is a property on the ViewModel

```
vm.dtInstance = {}
```

```
<table datatable="" dt-instance="ctrl.dtInstance" dt-options="ctrl.dtOptions" dt-columns="ctrl.dtColumns" class="table table-bordered table-striped hover"></table>
```

I have modified this so that it is no longer assigning the instance of the DataTable to **vm.dtInstance**. Instead I have declared a local variable within the controller and using a callback function to assign the DataTable instance to that local variable;

```
var dtInstance;

vm.dtInstanceCallback = function (dti) {
    dtInstance = dti;
}
```

```
<table datatable="" dt-instance="ctrl.dtInstanceCallback" dt-options="ctrl.dtOptions" dt-columns="ctrl.dtColumns" class="table table-bordered table-striped hover"></table>
```

Doing it this way still allows me to access the DataTable instance, through the local **dtInstance** varaible, which is what I needed. Where as before the Cyclic Object error resulted in the vm.dtInstance being undefined.
